src/back/spv/writer.rs

```diff
@@ -1894,12 +1899,16 @@ impl Writer {
                     instruction.add_operand(index_id);
                 }
 
-                if instruction.type_id != Some(result_type_id) {
+                let inst_type_id = instruction.type_id;
+                block.body.push(instruction);
+                if inst_type_id != Some(result_type_id) {
                     let sub_id = self.id_gen.next();
-                    let index_id = self.get_index_constant(0, &ir_module.types)?;
-                    let sub_instruction =
-                        Instruction::vector_extract_dynamic(result_type_id, sub_id, id, index_id);
-                    block.body.push(sub_instruction);
+                    block.body.push(Instruction::composite_extract(
+                        result_type_id,
+                        sub_id,
+                        id,
+                        &[0],
+                    ));
```

Comment:
Not really a pressing issue, but we aren't reusing this ID, and are using exactly the same code underneath here.

Reply:
Please clarify. Which ID we can re-use? What is the code underneath?

Reply:
Sorry for not clarifying. I cannot send the link with linenumber now (Github mobile app doesn't support it).

It is at line 2074. That is exactly the same code. Even with the old code (the vector extract dynamic) it was also duplicated.

I was suggesting that maybe it could be combined, but I am not sure if that is either possible or feasible to do in this situation.

Reply:
By "combined" do you mean moving the call to `composite_extract` to some place that is shared, or actually associating the same ID with the result and using this ID in multiple places?

Reply:
What I mean was both things that you are pointing out, as I believed that it was the same thing and thus can be shared, both by code and by id. But I think it is not shared, is it?

Reply:
There is a few `composite_extract` things: one in `AccessIndex`, one in `write_texture_coordinates`, one in `ImageLoad`, one in `ImageSample`, and one in the `ImageQuery`. I don't think there is a good way to share any of them.

Reply:
Sure, that is also fine by me, I was just wondering. Thanks for this rather lengthy discussion.